### ISSUE TITLE
Rocky 8 update from 8.9 to 8.10 seems break pcluster build-image

### ISSUE THREAD

**Issue by zeekus** [aws/aws-parallelcluster#6293]
**Required Info:**
 - AWS ParallelCluster version "version": "3.9.2"
 - Full cluster configuration: N/A building an AMI with pcluster build-image. 
     - base AMI: ami-02391db2758465a87
     - aws update OS component   
 - Cluster name: N/A Image Builder issue. 
 - Output of `pcluster describe-cluster` command: N/A
  

**Bug description and how to reproduce:**
- Update a standard Rocky 8.9 AMI by doing a 'yum update;yum upgrade' 
- run pcluster build image with updated OS.  
- The update transforms Rocky 8.9 into Rocky 8.10 and chefs says:
- - arn log details: Stdout: [2024-06-14T15:04:58+00:00] FATAL: RuntimeError: lustre[Install FSx options] (aws-parallelcluster-environment::install line 22) had an error: RuntimeError: FSx for Lustre is not supported in this Rocky Linux version 8.10, supported versions are >= 8.2

* Similar to this: https://github.com/aws/aws-parallelcluster/issues/6239

```
cat /etc/rocky-release
Rocky Linux release 8.9 (Green Obsidian)
[rocky@ip-10-148-64-97 ~]$ cat /etc/rocky-release
rocky-release           rocky-release-upstream
[rocky@ip-10-148-64-97 ~]$ cat /etc/os-release
NAME="Rocky Linux"
VERSION="8.9 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.9"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.9 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2029-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.9"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.9"
```

- After running an update, the OS version changes.
```
cat /etc/os-release
NAME="Rocky Linux"
VERSION="8.10 (Green Obsidian)"
ID="rocky"
ID_LIKE="rhel centos fedora"
VERSION_ID="8.10"
PLATFORM_ID="platform:el8"
PRETTY_NAME="Rocky Linux 8.10 (Green Obsidian)"
ANSI_COLOR="0;32"
LOGO="fedora-logo-icon"
CPE_NAME="cpe:/o:rocky:rocky:8:GA"
HOME_URL="https://rockylinux.org/"
BUG_REPORT_URL="https://bugs.rockylinux.org/"
SUPPORT_END="2029-05-31"
ROCKY_SUPPORT_PRODUCT="Rocky-Linux-8"
ROCKY_SUPPORT_PRODUCT_VERSION="8.10"
REDHAT_SUPPORT_PRODUCT="Rocky Linux"
REDHAT_SUPPORT_PRODUCT_VERSION="8.10"
```
Please update to the Chef code to handle Rocky Linux version updates more gracefully.

**Comment by himani2411** [comment]
Hi [zeekus](https://github.com/zeekus),

Thank you for Identifying the Bug. 

We are working on the Bug Fix. 
Are you currently blocked by this? If yes, we can try to find a workaround for you.

**Comment by zeekus** [comment]
Thanks. The work around is just not to run updates. It is not the most secure approach, but it works.

**Comment by zeekus** [comment]
> Hi [zeekus](https://github.com/zeekus),
> 
> Thank you for Identifying the Bug.
> 
> We are working on the Bug Fix. Are you currently blocked by this? If yes, we can try to find a workaround for you.



Thanks. The work around is just not to run updates. It is not the most secure approach, but it works.

**Comment by rmarable-flaretx** [comment]
Is there an ETA on a fix?

**Comment by himani2411** [comment]
Hi [rmarable-flaretx](https://github.com/rmarable-flaretx) and [zeekus](https://github.com/zeekus),

We dont have an ETA on the next release.

For a current workaround 

1. You can clone the [ParallelCluster v3.9.2](https://github.com/aws/aws-parallelcluster-cookbook/releases/tag/v3.9.2) cookbook.
2. Make changes in `cookbooks/aws-parallelcluster-environment/resources/lustre/lustre_redhat8.rb` and `cookbooks/aws-parallelcluster-environment/resources/lustre/lustre_rocky8.rb` as per the 2 PR's below
* https://github.com/aws/aws-parallelcluster-cookbook/pull/2764
* https://github.com/aws/aws-parallelcluster-cookbook/pull/2765
3. Push the changes to their cloned Cookbook repo or Upload the Cookbook in your S3 bucket.
4. Then use the below setting in their Image build Config
```
For Cookbook Repo
DevSettings:
  Cookbook:
    ChefCookbook: https://github.com/<UserName>/aws-parallelcluster-cookbook/tarball/<BranchName>


For S3:
DevSettings:
  Cookbook:
     ChefCookbook: s3://<bucket>/<path if any>/aws-parallelcluster-cookbook.tgz
```

**Comment by hanwen-pcluste** [comment]
This issue has been fixed by 3.10.0, which was released last Thursday.
Thank you for reporting the issue to us!
